protocol/contracts/leaser/src/leaser.rs

```diff
@@ -121,28 +121,36 @@ where
     })
 }
 
-pub(super) fn try_close_protocol<ProtocolsRegistryLoader>(
-    storage: &dyn Storage,
+pub(super) fn try_close_protocol<ProtocolsRegistryLoader, MsgFactory>(
+    storage: &mut dyn Storage,
+    new_lease_code: Code,
+    max_leases: MaxLeases,
+    migrate_msg: MsgFactory,
     protocols_registry: ProtocolsRegistryLoader,
     migration_spec: ProtocolContracts<MigrationSpec>,
     force: ForceClose,
 ) -> ContractResult<MessageResponse>
 where
+    MsgFactory: Fn(Addr) -> MigrateMsg,
     ProtocolsRegistryLoader: FnOnce(&dyn Storage) -> ContractResult<Addr>,
 {
     if force == ForceClose::No && has_lease(storage) {
```

Comment:
[Clippy; -W pedantic] Prefer using `matches!` when dealing with enums. This also removes the need for implementing `(Partial|)Eq`.
```suggestion
    if matches!(force, ForceClose::No) && has_lease(storage) {
```

Reply:
hm, interesting, could you point me to the clippy item for more details?

Reply:
It appears I was thinking of a different lint, so I'll leave it to your discretion.  
[`clippy::match_like_matches_macro`](https://rust-lang.github.io/rust-clippy/master/#/match_like_matches_macro)